source/logging/logging.rst

```diff
@@ -0,0 +1,380 @@
+has support for specifying logging expectations in tests.
+
+Motivation for Change
+=====================
+A common complaint from our support team is that they don't know how to easily get debugging
+information from drivers. Some drivers provide debug logging, but others do not. For drivers that
+do provide it, the log messages produced and the mechanisms for enabling debug logging are
+inconsistent.
+Although users can implement their own debug logging support via existing driver events (SDAM,
+APM, etc), this requires code changes. It is often difficult to quickly implement and deploy such
+changes in production at the time they are needed, and to remove the changes afterward.
+Additionally, there are useful scenarios to log that do not correspond to existing events.
+Standardizing on debug log messages that drivers produce and how to enable/configure logging will
+provide TSEs, CEs, and MongoDB users an easier way to get debugging information out of our drivers,
+facilitate support of drivers for our internal teams, and improve our documentation around
+troubleshooting.
+
+Design Rationale
+================
+
+-----------------------------
+Truncation of large documents
+-----------------------------
+
+1. Why have an option? 
+    We considered a number of approaches for dealing with documents of potentially very large size
+    in log messages, e.g. command documents, including 1) always logging the full document, 2) only
+    logging documents with the potential to be large when the user opts in, and 3) truncating large
+    documents by default, but allowing the user to adjust the maximum length logged. We chose the
+    third option as we felt it struck the best balance between concerns around readability and
+    usability of log messages. In the case where data is sufficiently small, the default behavior
+    will show the user the full data. In the case where data is large, the user will receive a
+    readable message with truncated data, but have the option to see more or all of the data.
+
+2. Why are the units for max document length flexible? 
+    String APIs vary across languages, and not all drivers will be able to easily and efficiently
+    truncate strings in the same exact manner. The important thing is that the option exists and
+    that its default value is reasonable, and for all possible unit choices (byte, code point, or
+    grapheme) we felt 1000 was a reasonable default. See
+    `here <https://exploringjs.com/impatient-js/ch_unicode.html>`_ for a helpful primer on related
+    Unicode concepts.
+
+--------------------------------------
+Structured versus Unstructured Logging
+--------------------------------------
+The MongoDB server produces structured logs as of 4.4, so it seems natural that MongoDB drivers
+might too. However, the idiomatic logging mechanisms of choice for some language ecosystems (e.g.
+Java) do not support structured logging yet, so we cannot require it.
```

Comment:
I'm not sure this is a strong argument.  Even though SLF4J doesn't support structured logging, the Java driver could still output log messages that _look_ like structured logs.  But still, it's not common to see logs like that in Java.

Reply:
Yeah, that works.

Reply:
updated